### PR TITLE
validator refactor

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: test/utils/importer_test.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            self.token,
+            mappings[str(params.get('file_format')).lower()].get('groups', []),
+            mappings[str(params.get('file_format')).lower()].get('date_columns', []),
+            mappings[str(params.get('file_format')).lower()].get('column_unit_regex', []),
+            {"samples": []},
+            header_row_index,
+            aliases
+        )
+
+        print('import samples output - ENIGMA')
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply test/utils/importer_test.py]: 👍

File: test/data/fake_samples_ENIGMA_1.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply test/data/fake_samples_ENIGMA_1.json:1]: yea. I updated them. the new one uses the new metadata names like `sesar:elevation_start`, `enigma:aquifer`, while the old one uses `elevation_start` and `aquifer`

File: test/core_tests.py [hunk 3]
```diff
@@ -160,7 +161,7 @@ def test_ENIGMA_file(self):
                 except:
                     raise ValueError(f"could not compare samples:\n{json.dumps(sample1)}\n{json.dumps(sample2)}")
 
-    # @unittest.skip('x')
+    @unittest.skip('x')
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply test/core_tests.py:164]: I wasn't sure. I don't think it should be skipped. I think someone might accidentally comment it out.

File: test/data/fake_samples.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
...
-]
\ No newline at end of file
+  {"id": "809c231a-0637-4c3b-b1fb-8cb233d26206", "user": "tgu2", "name": "s1", "node_tree": [{"id": "s1", "type": "BioReplicate", "parent": null, "meta_controlled": {"depth": {"value": 0, "units": "cm"}, "country": {"value": "Germany"}, "biome": {"value": "ENVO:01000221"}, "feature": {"value": "ENVO:01000002"}}, "meta_user": {"sesar:elevation_start": {"value": 123}, "country": {"value": "Germany"}, "feature": {"value": "ENVO:01000002"}, "biome": {"value": "ENVO:01000221"}, "depth": {"value": 0}, "sesar:material": {"value": "Gas"}, "elevation_unit": {"value": "cm"}}, "source_meta": [{"key": "depth", "skey": "Depth", "svalue": {"value": 0}}, {"key": "country", "skey": "Country", "svalue": {"value": "Germany"}}, {"key": "biome", "skey": "biome", "svalue": {"value": "ENVO:01000221"}}, {"key": "feature", "skey": "feature", "svalue": {"value": "ENVO:01000002"}}]}], "save_date": 1621458488627, "version": 1},
```
Comment (slebras) [inline test/data/fake_samples.json:2]: just noticed that this doesn't seem to group the `sesar:elevation_start` and `elevation_unit` fields.
they were previously grouped as: `{"units": "cm", "value": 123}`, but are now two separate entries

Comment (Tianhao-Gu) [reply test/data/fake_samples.json]: 💯

File: lib/sample_uploader/utils/importer.py [hunk 5]
```diff
@@ -24,6 +25,42 @@
 NOOP_VALS = ['ND', 'nd', 'NA', 'na', 'None', 'n/a', 'N/A', 'Na', 'N/a', '-']
 
 
+def find_header_row(sample_file, file_format):
```
Comment (slebras) [inline lib/sample_uploader/utils/importer.py:28]: could you add some brief documentation about what this function looks for when its trying to find the header row

File: lib/sample_uploader/utils/importer.py [hunk 6]
```diff
@@ -305,30 +386,20 @@ def import_samples_from_file(
...
+            df['sample_template'] = file_format.upper()
+
+        df, columns_to_input_names = format_input_file(df, params, {}, aliases)
 
         if not errors.get(severity='error'):
-            if params['file_format'].lower() in ['sesar', "enigma"]:
+            if file_format in ['sesar', "enigma"]:
                 if 'material' in df.columns:
-                    df.rename(columns={"material": params['file_format'].lower() + ":material"}, inplace=True)
+                    df.rename(columns={"material": file_format + ":material"}, inplace=True)
```
Comment (slebras) [inline lib/sample_uploader/utils/importer.py:399]: I think we can get rid of this section now that we have the aliasing. we can do that in another PR though

**Review by slebras — APPROVED** [review]
Looks great, just 1 comment really on expanding some documentation
